### PR TITLE
Pytorch 1.11 inference buildspec file changes 

### PR DESCRIPTION
**Pull request by RadhikaB-97**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pytorch/training/docker/1.11/py3/Dockerfile.cpu [hunk 1]
```diff
@@ -295,4 +295,4 @@ RUN HOME_DIR=/root \
  && rm -rf /tmp/tmp*
 
 ENTRYPOINT ["bash", "-m", "start_with_right_hostname.sh"]
-CMD ["/bin/bash"]
+CMD ["/bin/bash"]
```
Comment (shantanutrip) [inline pytorch/training/docker/1.11/py3/Dockerfile.cpu:298]: Is there a specific reason for making these changes?

Comment (RadhikaB-97) [reply pytorch/training/docker/1.11/py3/Dockerfile.cpu:298]: I haven't made any change, I just reverted it. Not sure why its appearing as a change.

Comment (shantanutrip) [reply pytorch/training/docker/1.11/py3/Dockerfile.cpu:298]: Sure, lets rectify it

File: pytorch/inference/docker/1.11/py3/cu115/Dockerfile.gpu [hunk 2]
```diff
@@ -164,12 +164,12 @@ RUN curl -L -o ~/mambaforge.sh https://github.com/conda-forge/miniforge/releases
...
     conda-content-trust \
     charset-normalizer \
- && /opt/conda/bin/conda install -c pytorch magma-cuda115=2.6.1 
+ && /opt/conda/bin/conda install -c pytorch magma-cuda115=2.6.1 \
 # Upstream conda looks to have moved to 4.13 which is incompatible with mamba 0.22.1 and will fail the conda-forge installs.
 # having "conda update conda" before the "conda -c conda-forge" commands will automatically update conda to 4.13.
 # Moving conda update conda" after the "conda -c conda-forge" commands keep conda at 4.12 but will update other packages using
 # the current conda 4.12
-RUN /opt/conda/bin/conda update -y conda \
+ && /opt/conda/bin/conda update -y conda \
```
Comment (shantanutrip) [inline pytorch/inference/docker/1.11/py3/cu115/Dockerfile.gpu:172]: Why are we doing this? Just to avoid a new step?

File: pytorch/buildspec.yml [hunk 3]
```diff
@@ -152,33 +152,33 @@ images:
...
+      <<: *INFERENCE_CONTEXT
+  BuildSageMakerGPUPTInferencePy3DockerImage:
+    <<: *INFERENCE_REPOSITORY
+    build: &PYTORCH_GPU_INFERENCE_PY3 false
+    image_size_baseline: 14000
+    base_image_name: BuildE3GPUPTInferencePy3DockerImage
+    device_type: &DEVICE_TYPE gpu
+    python_version: &DOCKER_PYTHON_VERSION py3
+    tag_python_version: &TAG_PYTHON_VERSION py38
+    cuda_version: &CUDA_VERSION cu115
```
Comment (shantanutrip) [inline pytorch/buildspec.yml:177]: I remember that e3 GPU images were released with cu115 and the SM GPU images were released with cu113, can you confirm please and make the changes if required?

File: pytorch/inference/docker/1.11/py3/cu115/Dockerfile.gpu (hunk 2 above)
Comment (RadhikaB-97) [reply pytorch/inference/docker/1.11/py3/cu115/Dockerfile.gpu:172]: New RUN command would create another image layer. It’s best to avoid doing that, as the clean command won’t be able to reduce the image size, because the clean step is at the next layer.

File: pytorch/buildspec.yml (hunk 3 above)
Comment (RadhikaB-97) [reply pytorch/buildspec.yml:177]: Yes, the sagemaker ones are released with cu113. Making the change and pushing a commit.

**Review by shantanutrip — DISMISSED** [review]
LGTM!! Revert the temp changes

**Review by shantanutrip — APPROVED** [review]
